### PR TITLE
Percentage display value fix for cohort tooltip

### PR DESCRIPTION
**Pull request by onursumer**
cohort_tooltip

# What? Why?
Fixes cBioPortal/cbioportal/issues/2691

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

File: src/shared/components/mutationTable/column/CohortColumnFormatter.tsx [hunk 1]
```diff
@@ -104,11 +104,16 @@ export default class CohortColumnFormatter {
         );
     }
 
-    private static getBoldPercentage(proportion:number) {
+    public static getBoldPercentage(proportion: number)
```
Comment (onursumer) [inline src/shared/components/mutationTable/column/CohortColumnFormatter.tsx]: we may also consider moving this function into a general purpose util class...

Comment (adamabeshouse) [inline src/shared/components/mutationTable/column/CohortColumnFormatter.tsx]: not a bad idea at all

Comment (onursumer) [inline src/shared/components/mutationTable/column/CohortColumnFormatter.tsx]: okay, now we have new functions in `FormatUtils`!

File: src/shared/lib/FormatUtils.ts [hunk 2]
```diff
@@ -13,3 +15,33 @@ export function toPrecision(value:number, precision:number, threshold:number)
...
 }
+
+/**
+ * Works the same as the default toFixed() function when the result string is not expected to be a zero value
+ * such as 0.0, 0.00, 0.000, etc.
+ *
+ * If the default toFixed() function returns a zero value, then converts it to an inequality such as
+ * <0.1, <0.01, <0.001, etc.
+ */
+export function toFixedWithThreshold(value: number, digits: number): string
```
Comment (onursumer) [inline src/shared/lib/FormatUtils.ts]: `toFixedWithThreshold` may not be the best name for this function

Comment (adamabeshouse) [inline src/shared/lib/FormatUtils.ts]: maybe just call it `toFixed` and have parameter `minBucket` ? or something like that

Comment (adamabeshouse) [inline src/shared/lib/FormatUtils.ts]: `toFixedWithThreshold` is fine with me too, but I think you should definitely make 0.1 a parameter

Comment (onursumer) [inline src/shared/lib/FormatUtils.ts]: `0.1` is dynamic actually. If the `digits` is 1 then it returns `<0.1`, so it is still "fixed" to one digit. If it is 2, returns `<0.01`, etc. This way, we still respect the `digits` parameter.

File: src/shared/components/cohort/FrequencyBar.tsx [hunk 3]
```diff
@@ -63,7 +74,7 @@ export default class FrequencyBar extends React.Component<IFrequencyBarProps, {}
                     textAnchor="start"
                     fontSize="10"
                 >
-                    {`${(100 * mainProportion).toFixed(1)}%`}
+                    {getPercentage(mainProportion)}
```
Comment (onursumer) [inline src/shared/components/cohort/FrequencyBar.tsx]: For consistency, using `getPercentage` function for the frequency bar as well...

File: src/shared/lib/FormatUtils.ts (hunk 2 above)
Comment (adamabeshouse) [inline src/shared/lib/FormatUtils.ts]: oh I see. I think sometimes you may want to specify a different threshold than `10^-(#digits)` though

**Pull request merged**